maker_dao/src/agent.ts

```diff
@@ -0,0 +1,163 @@
+import {
+  Finding,
+  Alert,
+  TransactionEvent,
+  FindingSeverity,
+  FindingType,
+  ethers,
+  getEthersProvider,
+  AlertsResponse,
+  AlertQueryOptions,
+} from "forta-agent";
+
+import {
+  L1_ESCROW_ADDRESS_OPTIMISM,
+  L1_ESCROW_ADDRESS_ARBITRUM,
+  L2_DAI_GATEWAY_ARB,
+  L1_AARBITRUM_GATEWAY,
+  TRANSFER_EVENT,
+  DEPOSIT_FINALISED_EVENT,
+  ETHER_CHAINID,
+  OP_CHAINID,
+  ARBI_CHAINID,
+  GET_DAI_BALANCE,
+  GET_TOTAL_SUPPLY,
+} from "./constant";
+
+const PROVIDER = getEthersProvider();
+
+const alert: Alert = {
+  alertId: "L1_ESCROW",
+  chainId: 1,
+  hasAddress: () => true,
+  metadata: {
+    totalSupply: Number,
+    network: "Ethereum",
+  },
+};
+
+
+const emptyAlertResponse: AlertsResponse = {
+  alerts: [alert],
+  pageInfo: { hasNextPage: false },
+};
+
+const query: AlertQueryOptions = { alertIds: ["L1_ESCROW"] };
+
+export const provideHandleTransaction =
+  (
+    provider: ethers.providers.JsonRpcProvider,
+    contractAddr: string,
+    getL1Alert: (alertQuery: AlertQueryOptions) => Promise<AlertsResponse>
+  ) =>
+  async (txEvent: TransactionEvent) => {
+    let findings: Finding[] = [];
+    const transferEvent = txEvent.filterLog(TRANSFER_EVENT);
+
+    const network = await provider.getNetwork();
+
+    const chainId = network.chainId;
+    const DepositEvent = txEvent.filterLog(DEPOSIT_FINALISED_EVENT);
+
+    if (chainId !== 1) {
+      for (const deposit of DepositEvent) {
+        const { l1Token, from, to, amount } = deposit.args;
+        const blockTag = txEvent.blockNumber;
+
+        if (chainId === OP_CHAINID || chainId === ARBI_CHAINID) {
+          const data = await GET_TOTAL_SUPPLY(contractAddr, provider, blockTag);
+
+          findings.push(
+            Finding.fromObject({
+              name: `DAI Bridge Deposit Transaction`,
+              description: `Deposit occur on DAI ${chainId} chainId bridge`,
+              alertId: `${
+                chainId === OP_CHAINID ? "OPTIMISM-1" : "ARBITRUM-1"
+              }`,
+              severity: FindingSeverity.Low,
+              type: FindingType.Info,
+              metadata: {
+                totalSupply: data.toString(),
+                amount: amount.toString(),
+                from: from,
+                to: to,
+              },
+            })
+          );
+        }
+      }
+    }
+
+    if (chainId === ETHER_CHAINID) {
+      if (transferEvent[0]) {
+        const toAddress = transferEvent[0].args?.to?.toLowerCase();
```

Comment:
```suggestion
      if (transferEvent[0]) {
        const transferEvent = txEvent.filterLog(TRANSFER_EVENT);
        const toAddress = transferEvent[0].args?.to?.toLowerCase();
```
Similar to the other suggestion, the filtering function could be moved here.

Reply:
```suggestion
      if (transferEvent[0]) {
        const toAddress = transferEvent[0].args!.to!.toLowerCase();
```
Also, this change to "make sure" those properties are present.